drivers/at_cmd/at_cmd.c

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <logging/log.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <net/socket.h>
+#include <init.h>
+#include <bsd_limits.h>
+
+#include <at_cmd.h>
+
+LOG_MODULE_REGISTER(at_cmd, CONFIG_AT_CMD_LOG_LEVEL);
+
+#define RESPONSE_MAX_LEN  CONFIG_AT_CMD_RESPONSE_MAX_LEN
+#define THREAD_PRIORITY   K_PRIO_PREEMPT(CONFIG_AT_CMD_THREAD_PRIO)
+
+#define AT_CMD_OK_STR    "OK"
+#define AT_CMD_ERROR_STR "ERROR"
+#define AT_CMD_CMS_STR   "+CMS:"
+#define AT_CMD_CME_STR   "+CME:"
+
+static K_THREAD_STACK_DEFINE(socket_thread_stack, 2048);
```

Comment:
Should this number not depend on `RESPONSE_MAX_LEN` ?
Or should the response buffer be `static`?

Reply:
I don't know of any good formula for calculating this number, I know that all of the variables in the thread will likely end up there, but then there is context/metadata that I don't have control over like the k_thread struct and maybe there are more data going on the stack? Maybe stack = variables + sizeof(k_thread)? Is this documented somewhere? I was looking in the zephyr documentation, but I could not find anything. 

I have tuned the number a bit down, and the at_client sample at least still seems happy, but I am not sure over time.